src/Query/Builder.php

```diff
@@ -1660,7 +1663,10 @@ private function aliasIdForResult(array|object $values): array|object
             }
 
             foreach ($values as $key => $value) {
-                if (is_array($value) || is_object($value)) {
+                if ($value instanceof UTCDateTime) {
+                    $values[$key] = Date::instance($value->toDateTime())
+                        ->setTimezone(new DateTimeZone(date_default_timezone_get()));
```

Comment:
I'll note that ODM also applies the default timezone in its [DateType class](https://github.com/doctrine/mongodb-odm/blob/2.8.1/lib/Doctrine/ODM/MongoDB/Types/DateType.php#L50).